### PR TITLE
[WIP] Issue #2143 Check default network state using libvirt

### PR DESCRIPTION
**Pull request by alaypatel07**
Fix [#2143](https://github.com/minishift/minishift/issues/2143)
Fix [#2121](https://github.com/minishift/minishift/issues/2121)

For libvirt-go bindings to work, libvirt-dev(el) packages needs to be installed.
Should I add `sudo apt-get install -y libvirt-dev` to before_install section in .travis.yml according to [this](https://docs.travis-ci.com/user/installing-dependencies/)

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
> For libvirt-go bindings to work, libvirt-dev(el) packages needs to be installed.
> Should I add sudo apt-get install -y libvirt-dev to before_install section in .travis.yml according to this

@alaypatel07 The issue is we should be able to compile Minishift on OSX , Linux and Windows , but if we use libvirt-go this will not be possible. In this case I won't use libvirt-go for Minishift as it does not bring considerable benefits or fixes which can not be solved otherwise. @minishift/minishift-dev  WDYT?

**Comment by gbraad** [comment]
You should have refered to issue #2143. Please make sure compilations will work on macOS, and at least compile using `make cross`

**Comment by gbraad** [comment]
Any updates?

**Comment by alaypatel07** [comment]
@gbraad I am tried the _linux.go way, but the make cross tool is not able to recognize libvirt-go bindings.

**Comment by praveenkumar** [comment]
I am putting this behind WIP so once every test pass we can remove WIP tag.

**Comment by budhrg** [comment]
@alaypatel07 Need rebase now.

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -340,26 +340,17 @@ func checkLibvirtDefaultNetworkExists() bool {
...
-	for _, stdOut = range outputSlice {
-		stdOut = strings.TrimSpace(stdOut)
-		match, err := regexp.MatchString("^default\\s", stdOut)
-		if err != nil {
-			return false
-		}
-		if match && strings.Contains(stdOut, "active") {
-			return true
-		}
+	defaultNet, err := conn.LookupNetworkByName("default")
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: :+1:

**Comment by budhrg** [comment]
@alaypatel07 Please rebase your PR.

**Comment by anjannath** [comment]
@alaypatel07 Any ETA  on when are you planning to fix the PR?

**Comment by praveenkumar** [comment]
I am closing this PR since there is no response from the @alaypatel07, please create new PR if you still want to work on that.

**Pull request closed without merging**